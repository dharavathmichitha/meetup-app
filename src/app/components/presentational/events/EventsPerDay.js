import React from 'react';
import PropTypes from 'prop-types';
import EventContainer from '../../container/EventContainer';

const EventsPerDay = ({ date, events }) => {
  const eventsCollection = events.map(event => {
    const key = `key_${event.id}`;
    return <EventContainer key={key} event={event} />;
  });
  return (
    <>
      <header>
        <h2>{date}</h2>
      </header>
      <div role="listitem">
        <div className="event-list-perday" role="list">
          {eventsCollection}
        </div>
      </div>
    </>
  );
};

EventsPerDay.propTypes = {
  date: PropTypes.string.isRequired,
  events: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      isFree: PropTypes.bool.isRequired,
      name: PropTypes.string.isRequired,
      city: PropTypes.number.isRequired,
      startDate: PropTypes.string.isRequired,
      endDate: PropTypes.string.isRequired,
    }),
  ).isRequired,
};

export default EventsPerDay;
