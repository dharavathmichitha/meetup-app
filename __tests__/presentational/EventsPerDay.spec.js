import React from 'react';
import renderer from 'react-test-renderer';
import EventsPerDay from '../../src/app/components/presentational/events/EventsPerDay';

beforeEach(() => {
  jest.resetModules();
});

describe('EventsPerDay Component', () => {
  it('should match snapshot including default props', () => {
    const firstMondayJuneEvent = {
      id: 0,
      isFree: false,
      name: 'firstTestEvent',
      city: 1,
      startDate: '2019-07-14T15:00:00+00:00',
      endDate: 'Thu Jan 01 1970',
    };
    const secondMondayJuneEvent = {
      id: 1,
      isFree: false,
      name: 'secondTestEvent',
      city: 1,
      startDate: '2019-07-14T17:00:00+00:00',
      endDate: 'Thu Jan 01 1970',
    };
    const dayEvents = [firstMondayJuneEvent, secondMondayJuneEvent];
    const props = {
      date: 'Monday 29th June',
      events: dayEvents,
    };
    const component = renderer.create(<EventsPerDay {...props} />);

    const componentTree = component.toJSON();

    expect(componentTree).toMatchSnapshot();
  });
});
