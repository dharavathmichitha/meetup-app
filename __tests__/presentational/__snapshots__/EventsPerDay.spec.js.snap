// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventsPerDay Component should match snapshot including default props 1`] = `
Array [
  <header>
    <h2>
      Monday 29th June
    </h2>
  </header>,
  <div
    role="listitem"
  >
    <div
      className="event-list-perday"
      role="list"
    >
      <div
        className="event"
        role="listitem"
      >
        <div
          aria-level="3"
          className="event__time"
          role="heading"
        >
          18.00
        </div>
        <div
          className="event__info"
        >
          <div
            aria-level="4"
            className="event__title"
            role="heading"
          >
            firstTestEvent
            <span
              className="event__title--free"
            />
          </div>
          <div
            className="event__info__city-name"
          >
             - -434201h
          </div>
        </div>
        <button
          className="event__sign-up btn"
          onClick={[Function]}
          type="button"
        >
          Sign up
        </button>
      </div>
      <div
        className="event"
        role="listitem"
      >
        <div
          aria-level="3"
          className="event__time"
          role="heading"
        >
          20.00
        </div>
        <div
          className="event__info"
        >
          <div
            aria-level="4"
            className="event__title"
            role="heading"
          >
            secondTestEvent
            <span
              className="event__title--free"
            />
          </div>
          <div
            className="event__info__city-name"
          >
             - -434203h
          </div>
        </div>
        <button
          className="event__sign-up btn"
          onClick={[Function]}
          type="button"
        >
          Sign up
        </button>
      </div>
    </div>
  </div>,
]
`;
